File: docs/app/lib/gate.js
// Legal & ethics gate — mirrors osint_omega.gate.LegalEthicsGate
import { TargetType } from "./detect.js";

const PERSONAL_TYPES = new Set([
  TargetType.EMAIL, TargetType.PHONE, TargetType.PERSON, TargetType.USERNAME,
]);

export function evaluateGate(target, scope) {
  const notes = [];
  let allowed = true;
  let restricted = false;

  if (scope === "LEGALLY_RESTRICTED") {
    allowed = false;
    notes.push("Scope LEGALLY_RESTRICTED : aucune collecte autorisée.");
    return { allowed, restricted, notes };
  }

  if (PERSONAL_TYPES.has(target.type) && (scope === "SANDBOX_TEST" || scope === "PUBLIC_INTEREST_RESEARCH")) {
    restricted = true;
    notes.push("Donnée personnelle : appliquer la minimisation (RGPD art. 5). Justifier l'intérêt légitime et documenter la base légale.");
  }

  if (target.type === TargetType.ONION) {
    restricted = true;
    notes.push("Cible .onion : utiliser Tor/Whonix, ne pas dé-anonymiser sans mandat.");
  }

  if (target.type === TargetType.PERSON && scope !== "CLIENT_AUTHORIZED_SCOPE" && scope !== "INTERNAL_AUDIT") {
    restricted = true;
    notes.push("Identification d'une personne physique : vérifier le consentement ou l'intérêt public.");
  }

  if (scope === "SANDBOX_TEST") {
    notes.push("Mode sandbox : résultats à des fins pédagogiques uniquement, ne pas publier.");
  }

  return { allowed, restricted, notes };
}

File: osint_omega/osint_omega/gate.py
"""LegalEthicsGate — barrière de conformité en amont de toute collecte.

Inspirée du prompt multi-agent forensique : chaque mission doit être classée
dans un périmètre explicite et est refusée/restreinte sinon.
"""

from __future__ import annotations

from dataclasses import dataclass

from osint_omega.types import Scope, Target, TargetType


@dataclass(frozen=True, slots=True)
class GateDecision:
    allowed: bool
    restricted: bool
    notes: tuple[str, ...]

    @property
    def status_text(self) -> str:
        if not self.allowed:
            return "REFUSED"
        return "RESTRICTED" if self.restricted else "ALLOWED"


class LegalEthicsGate:
    """Filtre les missions hors cadre et ajoute des notes de minimisation."""

    PERSONAL_TYPES = frozenset({TargetType.EMAIL, TargetType.PERSON, TargetType.USERNAME})

    def evaluate(self, target: Target, scope: Scope) -> GateDecision:
        notes: list[str] = []

        if scope == Scope.LEGALLY_RESTRICTED:
            notes.append(
                "Périmètre LEGALLY_RESTRICTED : seule une simulation / "
                "une méthodologie peut être produite ; aucune collecte réelle."
            )
            return GateDecision(allowed=False, restricted=True, notes=tuple(notes))

        if target.type in self.PERSONAL_TYPES and scope in {
            Scope.SANDBOX_TEST,
            Scope.PUBLIC_INTEREST_RESEARCH,
        }:
            notes.append(
                "Cible personnelle hors périmètre mandaté : minimisation "
                "obligatoire, ne pas attribuer d'identité réelle sans base forte."
            )

        if target.type == TargetType.ONION:
            notes.append(
                "Cible .onion : activer Tor, journaliser chaque requête, "
                "ne jamais télécharger de contenu illégal."
            )

        if scope == Scope.SANDBOX_TEST:
            notes.append("SANDBOX_TEST : résultats à considérer comme non probatoires.")

        return GateDecision(allowed=True, restricted=False, notes=tuple(notes))
